rust/main/chains/hyperlane-ethereum/src/contracts/mailbox.rs

```diff
@@ -547,8 +554,11 @@ where
         &self,
         message: &HyperlaneMessage,
         metadata: &[u8],
+        apply_gas_overrides: bool,
     ) -> ChainResult<TxCostEstimate> {
-        let contract_call = self.process_contract_call(message, metadata, None).await?;
+        let contract_call = self
+            .process_contract_call(message, metadata, apply_gas_overrides, None)
```

Comment:
I think if `apply_gas_overrides` is false, when we call `process_contract_call` we end up never calling eth_estimateGas, and we'd hit the `.ok_or(HyperlaneProtocolError::ProcessGasLimitRequired)?;` - I assume this is why e2e is failing